### DISCUSSION TITLE
How can I get Entries with a certain Taxonomy Term attached in Laravel?

### DISCUSSION THREAD

**Discussion by mikemike** [statamic/cms#6278] [Troubleshooting]
I'm using the `Entry` Facade to get entries from Statamic in Laravel (I'm not using Antlers). I have a simple concept of News posts with Categories. When a user selects a category I want to get all entries that are tagged with that category.

My collection is called 'news'.
My Taxonomy is called 'news_categories'.
This is attached to the collection using a field called 'categories'.

Below is my code:

```
public function index(Request $request): View
{
    $newsObj = Entry::query()
        ->where('collection', 'news');

    if($request->category) {
        $term = Term::findBySlug($request->category, 'news_category'); // news_category::business-news
        
        $newsObj = $newsObj->whereTaxonomy($term->id);
    }
    
    $news = $newsObj->paginate();

    return view('global.news.index', [
        'news' => $news
    ]);
}
```

This returns an empty collection every time a category is selected (eg whenever `$request->category` is set).

I can't find any documentation on `whereTaxonomy` so I've had to go digging in Statamic's code to find it. 

Any help greatly appreciated.

**Comment by ryanmitchell** [comment]
Terms dont have ids, they have slugs... so you should be able to do `$newsObj->whereTaxonomy('news_category::term_slug')`

Alternatively you can come at it the other way, and get the Term using the Term::query facade, and then get the entries of the term.

**Reply by mikemike** [reply]
Thanks @ryanmitchell , but in the above code example `$term->id` is actually the slug (that's that `Term` gives me). The comment next to it is the actual value it returns, in this case `news_category::business-news`. I've tried hard-coding it too, but the same result.

I might have to go with your alternative approach, although I'd really like to solve this one.

**Reply by edalzell** [reply]
Why not use the `Statamic` helper in blade?
